lib/core/msdf-sys/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+            &JsValue::from_str(emscripten_data_types::NUMBER)
+        );
+        unsafe { // Note [Usage of Uint8Array::view]
+            let params = js_sys::Array::of2(
+                &JsValue::from(Uint8Array::view(data)),
+                &JsValue::from_f64(data.len() as f64)
+            );
+            let handle = emscripten_call_function(
+                "msdfgen_loadFontMemory",
+                emscripten_data_types::NUMBER,
+                param_types,
+                params);
+            Font { handle }
+        }
+    }
+}
+
+/*
+ * Note [Usage of Uint8Array::view]
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * We use view in this place to avoid copying font data. This is the only way
+ * to do it with js_sys structures. The Uint8Array does not leave function
+ * scope, so does not excess lifetime of data
+ */
+
+impl Drop for Font {
+    fn drop(&mut self) {
+        msdfgen_free_font(self.handle.clone())
+    }
+}
+
+// =======================================
+// === MutlichannelSignedDistanceField ===
+// =======================================
+
+// Mutlichannel Signed Distance Field for one glyph
+//
+// For more information about MSDF see https://github.com/Chlumsky/msdfgen
+pub struct MutlichannelSignedDistanceField {
+    pub width  : usize,
+    pub height : usize,
+    pub data   : [f32; Self::MAX_DATA_SIZE]
+}
+
+// Parameters of MSDF generation
+//
+// The structure gathering MSDF generation parameters meant to be same for all rendered glyphs
+pub struct MSDFParameters {
+    pub width                         : usize,
+    pub height                        : usize,
+    pub edge_coloring_angle_threshold : f64,
+    pub range                         : f64,
+    pub edge_threshold                : f64,
+    pub overlap_support               : bool
+}
+
+impl MutlichannelSignedDistanceField {
+    pub const MAX_SIZE       : usize = 32;
+    pub const CHANNELS_COUNT : usize = 3;
+    pub const MAX_DATA_SIZE  : usize = Self::MAX_SIZE * Self::MAX_SIZE *
+        Self::CHANNELS_COUNT;
+
+    pub fn generate(
+        font      : &Font,
+        unicode   : u32,
+        params    : MSDFParameters,
+        scale     : Vector2D<f64>,
+        translate : Vector2D<f64>,
+    ) -> MutlichannelSignedDistanceField {
+        assert!(params.width  <= Self::MAX_SIZE);
+        assert!(params.height <= Self::MAX_SIZE);
+
+        let output_address = msdfgen_generate_msdf(
+            params.width,
+            params.height,
+            font.handle.clone(),
+            unicode,
+            params.edge_coloring_angle_threshold,
+            params.range,
+            scale.x,
+            scale.y,
+            translate.x,
+            translate.y,
+            params.edge_threshold,
+            params.overlap_support
+        );
+        let mut data : [f32; Self::MAX_DATA_SIZE] = [0.0; Self::MAX_DATA_SIZE];
+        let data_size = params.width*params.height*Self::CHANNELS_COUNT;
```

Comment:
Again very minor thing - spaces around operators. Damn, we need some day to fix rustfmt to support our style. Its so buggy now its hard to be used, but comments like these coulb be tiresome for you, sorry Adam.

Reply:
No problem. fixed